x/payment/types/builder.go

```diff
@@ -44,6 +47,8 @@ func (k *KeyringSigner) QueryAccountNumber(ctx context.Context, conn *grpc.Clien
 		return err
 	}
 
+	k.Lock()
+	defer k.Unlock()
```

Comment:
We should not lock the mutex for the entire network request inside `QueryAccount`. Instead, the lock should be applied to the read/write of shared memory.